src/Support/SupportData.php

```diff
@@ -0,0 +1,930 @@
+		$error_data = json_decode( $error_data, true );
+
+		ksort( $error_data );
+
+		/**
+		 * Generate new slug after removing site specific data.
+		 */
+		$error_data['error_slug'] = static::generate_hash( $error_data );
+
+		return $error_data;
+	}
+
+	/**
+	 * To normalize the error source data.
+	 *
+	 * @since 2.2
+	 *
+	 * @param array $source Error source detail.
+	 *
+	 * @return array Normalized error source data.
+	 */
+	public static function normalize_error_source( $source ) {
+
+		if ( empty( $source ) || ! is_array( $source ) ) {
+			return [];
+		}
+
+		static $plugin_versions = [];
+		static $theme_versions  = [];
+
+		/**
+		 * All plugin info
+		 */
+		if ( empty( $plugin_versions ) || ! is_array( $plugin_versions ) ) {
+
+			$plugin_list = get_plugins();
+			$plugin_list = array_keys( $plugin_list );
+			$plugin_list = array_values( array_unique( $plugin_list ) );
+			$plugin_list = array_map( __CLASS__ . '::normalize_plugin_info', $plugin_list );
+
+			foreach ( $plugin_list as $plugin ) {
+				$plugin_versions[ $plugin['slug'] ] = $plugin['version'];
+			}
+		}
+
+		/**
+		 * All theme info.
+		 */
+		if ( empty( $theme_versions ) || ! is_array( $theme_versions ) ) {
+
+			$theme_list = array_merge( [ wp_get_theme() ], wp_get_themes() );
+
+			foreach ( $theme_list as $theme ) {
+				if ( ! empty( $theme ) && is_a( $theme, 'WP_Theme' ) ) {
+					$theme_versions[ $theme->get_stylesheet() ] = $theme->get( 'Version' );
+				}
+			}
+		}
+
+		/**
+		 * Normalize error source.
+		 */
+
+		$allowed_types  = [ 'plugin', 'theme' ];
+		$source['type'] = ( ! empty( $source['type'] ) ) ? strtolower( trim( $source['type'] ) ) : '';
+
+		/**
+		 * Do not include wp-core sources.
+		 */
+		if ( empty( $source['type'] ) || ! in_array( $source['type'], $allowed_types, true ) ) {
+			return [];
+		}
+
+		if ( 'plugin' === $source['type'] ) {
+			$source['version'] = $plugin_versions[ $source['name'] ];
+		} elseif ( 'theme' === $source['type'] ) {
+			$source['version'] = $theme_versions[ $source['name'] ];
+		}
+
+		if ( ! empty( $source['text'] ) ) {
+			$source['text'] = trim( $source['text'] );
+			$source['text'] = static::remove_domain( $source['text'] );
+		}
+
+		// Generate error source slug.
+		$error_source_slug = self::generate_hash( $source );
+
+		// Update source information. Add error_slug and source_slug.
+		$source['error_source_slug'] = $error_source_slug;
+
+		ksort( $source );
+
+		return $source;
+	}
+
+	/**
+	 * To get amp validated URLs.
+	 *
+	 * @since 2.2
+	 *
+	 * @return array List amp validated URLs.
+	 */
+	public function get_amp_urls() {
+
+		global $wpdb;
+
+		$query = "SELECT ID, post_title, post_content FROM {$wpdb->posts} WHERE `post_type`='amp_validated_url'";
+
+		if ( ! empty( $this->urls ) && is_array( $this->urls ) ) {
+			$placeholder = implode( ', ', array_fill( 0, count( $this->urls ), '%s' ) );
+			$query      .= ' AND post_title IN ( ' . $placeholder . ' ) ';
```

Comment:
Instead of using the `post_title` and constructing a SQL string, the `post_name` field is an md5 hash of the URL. Therefore, `WP_Query` could be used instead along with `post_name__in`.